components/query/feature.component.ts

```diff
@@ -0,0 +1,62 @@
+import {Component, OnInit} from '@angular/core';
+import {HsMapService} from '../map/map.service';
+import {HsQueryVectorService} from './query-vector.service';
+import {Input} from '@angular/core';
+
+@Component({
+  selector: 'hs-query-feature',
+  template: require('./partials/feature.html'),
+})
+export class HsQueryFeatureComponent implements OnInit {
+  @Input() feature;
+  attributeName = '';
+  attributeValue = '';
+  newAttribVisible = false;
+  exportFormats = [{name: 'WKT format'}];
+  constructor(
+    private HsMapService: HsMapService,
+    private HsQueryVectorService: HsQueryVectorService
+  ) {}
+
+  ngOnInit(): void {}
+
+  olFeature() {
+    return this.feature.feature;
+  }
+
+  isFeatureRemovable() {
+    if (this.feature.feature) {
+      return this.HsQueryVectorService.isFeatureRemovable(this.olFeature());
+    } else {
+      return false;
+    }
+  }
+
+  saveNewAttribute(attributeName, attributeValue) {
+    if (this.feature && this.feature.feature) {
```

Comment:
Will be using `this.feature.feature !== undefined` instead

Reply:
I don't recommend that. It would throw an error when `this.feature` equals undefined or null.

Reply:
Hmm... I did try Your suggestion,(`this.feature?.feature`) yet it returned undefined as well

Reply:
Probably should leave this like it is, cannot think of any better check  right now. <?.> in not working

Reply:
Maybe we should also target es2019 as per https://stackoverflow.com/questions/58725711/how-to-get-optional-chaining-working-in-typescript . In case we want to drop babel in future.

Reply:
The thing is, we already use optional chaining in our TS files (e.g. https://github.com/hslayers/hslayers-ng/blob/develop/components/layermanager/layermanager.component.ts#L97) and it works like a charm. Targeting es5 is a good setting in my opinion. It compiles the TS code into a pretty basic JS, which will secure that hslayers will run (somehow) even in older browsers. Which might be important for mobile browsers for instance.

@DailisLangovskis If it is causing you too much pain (as I can see it does :-) ) just drop my suggestion, which is more or less stylistic and go on. This condition is no game changer. There are more important things to do I believe.

Reply:
@jmacura Right, I thought that we were targeting esnext but it was in the lib option in tsconfig. Didn't check first. In that case I onder hy it didnt work before, because we are having the chaining in ts file. Will have to make a test for this so we don't break the optional chaining in future with some tsconfig/webpack change.

Reply:
@jmacura I am not bothered by this what so ever, just wanted to make sure, it if working fine. Now I only want to understand, do we need the babel or not? I just did this in my spare to for educational purposes :)

Reply:
@DailisLangovskis As far as I can tell, we only need Babel for the rest of the JS files, until we rewrite them to TS. TypeScript compiler (or `ts-loader` in our case) will do the job afterwards.